services/telemetry/src/util.js

```diff
@@ -19,37 +19,37 @@ export function cleanupAfter(cleanup, limit, message) {
 
   let earlyCall = false;
   let earlyErr = null;
+  let cleanupObj = {};
```

Comment:
So in the end, why did this have to be done?

Reply:
In the case where, 
```
let wait = async () => await new Promise((resolve, reject) => {
    // Call the finish function if the timeout is reached.
    let waitTimeout = setTimeout(() => {
      finish(Error(timeoutMessage));
    }, limit);

    // Cancel the timeout, cleanup, and mark the operation as
    // complete.
    finish = (err, result) => {
      clearTimeout(waitTimeout);
      cleanup();

      if (err) reject(err);
      else resolve(result);
    };

    if (earlyCall) finish(earlyErr);
  });
return { finish, wait };
```
Overriding the "finish" function when "wait" is called doesn't actually update the function where "cleanupAfter" returned.  The variable is only updated in cleanupAfter's closure. I think that's why the tests were failing initially.

Reply:
Oh, I see, because of pass-by-value.